data/trie/evictionWaitingList/evictionWaitingList.go

```diff
@@ -92,3 +98,32 @@ func (ewl *evictionWaitingList) IsInterfaceNil() bool {
 func (ewl *evictionWaitingList) GetSize() uint {
 	return ewl.cacheSize
 }
+
+// PresentInNewHashes searches for the given hash in all of the evictionWaitingList's newHashes
+func (ewl *evictionWaitingList) PresentInNewHashes(hash string) (bool, error) {
+	for key := range ewl.cache {
+		lastByte := key[len(key)-1]
```

Comment:
Could be len(key) = 0? If yes, you should add a check here

Reply:
len(key) should never be 0, but added the check anyway.